### PR TITLE
refactoring of health checks 

### PR DESCRIPTION
**Pull request by SchlottiP**
- "/health/" returns "OK" and 200, if all required services are healthy, "ERR" and "503" if not

- "/health/check" returns two lists. One for the required and one for the optional checks, the list shows the name, the state and a description of the state for each registered health check.

- "/health/liveness" returns always "OK" and 200

- "/health/readiness" is configureable with `healthReadinessCheck(func(http.ResponseWriter, *http.Request)) `, default: returns "OK" and 200

### CODE REVIEW

File: maintenance/health/servicehealthcheck/readable_health_handler.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package servicehealthcheck
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+// saves length of the longest name for the column width in the table. 20 characters width is the default
+var longestCheckName = 20
```
Comment (daemonfire300) [inline maintenance/health/servicehealthcheck/readable_health_handler.go]: If this a package internal value and we have no way of configuring (did I miss something) then we can make this a `const`, can't we?

File: maintenance/health/servicehealthcheck/readable_health_handler.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+func (h *readableHealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	reqChecks := check(&requiredChecks)
+	optChecks := check(&optionalChecks)
+
+	status := http.StatusOK
+	table := "%-" + strconv.Itoa(longestCheckName) + "s   %-3s   %s\n"
+	body := "Required Services: \n"
+	for name, res := range reqChecks {
```
Comment (daemonfire300) [inline maintenance/health/servicehealthcheck/readable_health_handler.go]: Suggestion: https://golang.org/pkg/strings/#Builder

File: maintenance/health/servicehealthcheck/readable_health_handler.go (hunk 1 above)
Comment (SchlottiP) [reply maintenance/health/servicehealthcheck/readable_health_handler.go]: The Value is changed for each new registered health check (in `RegisterHealthCheck` and `RegisterOptionalHealthCheck`)

Comment (daemonfire300) [reply maintenance/health/servicehealthcheck/readable_health_handler.go]: Sorry, I must have misread the comment. It's explained right there :D

File: maintenance/health/servicehealthcheck/healthchecker.go [hunk 3]
```diff
@@ -4,85 +4,101 @@
...
-		h.writeError(w, errors.New("health check not registered"), http.StatusNotFound, name)
-		return
-	}
-	hc := hcInterface.(HealthCheck)
-
-	// If it was not possible to initialise this health check, then show the initialisation error
-	if val, isIn := initErrors.Load(name); isIn {
-		err := val.(error)
+const (
+	// Err State of a service, if a error occurred during the health check of the service
```
Comment (daemonfire300) [inline maintenance/health/servicehealthcheck/healthchecker.go]: `an error`

**Pull request merged**